images/snapshot-release-tags/snapshot_release_tags.py

```diff
@@ -59,4 +87,64 @@
         ref=f"refs/tags/{tag_name}",
         sha=tag.sha)
 
+    #####################################
+    # Setup the package repo to update the sha
+    spackages = "spack-packages"
+    os.makedirs(spackages)
+
+    branch_id = f"snapshot-update/{tag.name}"
+    with ctx.chdir(spackages):
+        _durable_subprocess_run(["git", "init"])
+        _durable_subprocess_run(["git", "config", "user.email", "noreply@spack.io"])
+        _durable_subprocess_run(["git", "config", "user.name", "spackbot"])
+        _durable_subprocess_run(["git", "config", "advice.detachedHead", "false"])
+        _durable_subprocess_run(["git", "remote", "add", "github", f"git@github.com:{GITHUB_REPO}"])
+        _durable_subprocess_run(["git", "fetch", "--depth", "1", "develop"])
+        _durable_subprocess_run(["git", "checkout", "-b", branch_id])
+
+    gitlab_ci_yml = os.path.join(spackages, ".ci", "gitlab", ".gitlab-ci.yml")
+    shasum_re = re.compile(r"SPACK_CHECKOUT_VERSION:( )*\"([a-z0-9]{40})\"")
+    try:
+        # Update the commit if possible
+        with tempfile.TemporaryFile() as tmp:
+            shutil.copy(gitlab_ci_yml, tmp)
+            with open(tmp, "r") as ifd, open(gitlab_ci_yml, "w") as ofd:
+                for line in ifd:
+                    old_sha_match = re.search(shasum_re, line)
+                    if old_sha_match:
+                        old_sha = old_sha_match[2]
+                        # The if shas match, don't update
+                        if old_sha == tag.sha:
+                            raise RuntimeError("sha has not changed")
+                        old_date = py_gh_repo.get_commit(old_sha).commit.committer.date
+                        new_date = py_gh_repo.get_commit(tag.sha).commit.committer.date
+                        # The if new sha is older than the old sha, don't update
+                        if old_date > new_date:
+                            raise RuntimeError("snapshot sha is older than current sha")
+                        line = re.sub(shasum_re, line, tag.sha)
+                    ofd.write(line)
+
+            with ctx.chdir(spackages):
+                commit_msg = f"""Snapshot Sync: {tag.name} {tag.sha}"""
+                _durable_subprocess_run(["git", "commit", "-a", "-m", commit_msg])
+                _durable_subprocess_run(["git", "push", "-u", "github", ])
+
+            spackages_repo = py_github.get_repo("spack/spack-packages", lazy=True)
+            # Create the PR
+            pr = spackages_repo.create_pull(
+                base="develop",
+                head=branch_id,
+                body="""
+Automated sync commit created by the snapshot release tags cron.
+
+@spack/maintainers
+"""
+            )
+            last_commit = pr.get_commits()[pr.commits - 1]
+            # Trigger a rebuild everything on the PR to be reviewed by maintainers
+            pr.create_comment("@spackbot rebuild everything", last_commit, "", 0)
```

Comment:
I think this should work with the urgent tag, but I am not sure about timing of the events sent to spackbot and the periodicity of the sync script.

Reply:
Use a label to denote a different branch name scheme to use for this type of CI.